src/azure-cli/azure/cli/__main__.py

```diff
@@ -74,3 +74,25 @@ def cli_main(cli, args):
     telemetry.set_init_time_elapsed("{:.6f}".format(init_finish_time - start_time))
     telemetry.set_invoke_time_elapsed("{:.6f}".format(invoke_finish_time - init_finish_time))
     telemetry.conclude()
+    try:
+        if sys.argv[1] != 'upgrade' and az_cli.config.getboolean('auto-upgrade', 'enable', False):
```

Comment:
Why do we put this at the end of CLI executing ? If the user set `auto-upgrade` and not prompt config, however when he run a command, CLI will execute with the old version then upgrade it, right ?

Reply:
Yes. We don't want the user to wait for the upgrade when typing the first command. 

With the prompt, users can check the result of the command first and then type yes to continue or no to decline the upgrade if they would like to run a second command without the wait for upgrade.

With no prompt config, I have suppressed the stdout, so it won't fill up the screen to block the actual result of the command. Besides, if the command result is assigned to a variable, the variable would only contain the actual command result in this case.